src/requirements/reqs-functions.ts

```diff
@@ -8,6 +9,121 @@ import {
   GroupedRequirementFulfillmentReport,
 } from './types';
 
+type FulfillmentStatistics = {
+  readonly id: string;
+  readonly requirement: RequirementWithIDSourceType;
+  readonly courses: readonly (readonly CourseTaken[])[];
+} & RequirementFulfillmentStatistics;
+
+/**
+ * @param course course object with useful information retrived from Cornell courses API.
+ * @returns true if the course is AP/IB equivalent course or credit
+ */
+const courseIsAPIB = (course: CourseTaken): boolean =>
+  [CREDITS_COURSE_ID, FWS_COURSE_ID].includes(course.courseId) ||
+  ['AP', 'IB', 'CREDITS'].includes(course.subject);
+
+/**
+ * Used for total academic credit requirements for all colleges except EN and AR
+ * @param course course object with useful information retrived from Cornell courses API.
+ * @returns true if the course is not PE or 10** level
+ */
+const courseIsAllEligible = (course: CourseTaken): boolean =>
+  course.courseId === CREDITS_COURSE_ID ||
+  (!courseIsAPIB(course) && course.subject !== 'PE' && !course.number.startsWith('10'));
+
+const getTotalCreditsFulfillmentStatistics = (
+  college: string,
+  courses: readonly CourseTaken[]
+): FulfillmentStatistics | null => {
+  const requirementCommon = {
+    sourceType: 'College',
+    sourceSpecificName: college,
+    name: 'Total Academic Credits',
+    courses: [],
+    subRequirementProgress: 'any-can-count',
+    fulfilledBy: 'credits',
+    minCount: 120,
+  } as const;
+  let requirement: RequirementWithIDSourceType;
+  switch (college) {
+    case 'AG':
+      requirement = {
+        ...requirementCommon,
+        id: 'College-AG-total-credits',
+        description:
+          '120 academic credits are required for graduation. ' +
+          'A minimum of 100 credits must be in courses for which a letter grade was recieved. ' +
+          'PE courses do not count.',
+        source: 'http://courses.cornell.edu/content.php?catoid=41&navoid=11561',
+      };
+      break;
+    case 'AS':
+      requirement = {
+        ...requirementCommon,
+        id: 'College-AS-total-credits',
+        description:
+          '120 academic credits are required' +
+          'PE courses and courses numbered 1000-1099 do not count towards the 120 credits',
+        source: 'http://courses.cornell.edu/content.php?catoid=41&navoid=11570#credit-req',
+      };
+      break;
+    case 'HE':
+      requirement = {
+        ...requirementCommon,
+        id: 'College-HE-total-credits',
+        description:
+          '120 academic credits are required' +
+          'PE courses and courses numbered 1000-1099 do not count towards the 120 credits',
```

Comment:
I think some of the concatenated strings in the descriptions for the different colleges are missing periods and spaces at the end.

Reply:
fixed